src/main/java/com/cleanroommc/groovyscript/compat/mods/advancedrocketry/BaseRegistry.java

```diff
@@ -0,0 +1,209 @@
+        recipes.addAll(restoreFromBackup());
+        removeScripted().forEach(r -> recipes.removeIf(r::equals));
+    }
+
+    public void addRecipe(IRecipe r) {
+        getRecipeList().add(r);
+        addScripted(r);
+    }
+
+    public boolean removeByFluidInput(FluidStack fluidStack) {
+        return getRecipeList().removeIf(r -> {
+            List<FluidStack> inputFluid = r.getFluidIngredients();
+            if (inputFluid.stream().anyMatch(fluidIn -> fluidIn.isFluidEqual(fluidStack))) {
+                addBackup(r);
+                return true;
+            }
+            return false;
+        });
+    }
+
+    public boolean removeByInput(IIngredient inputItem) {
+        if (inputItem instanceof FluidStack) {
+            return removeByFluidInput((FluidStack) inputItem);
+        }
+        return getRecipeList().removeIf(r -> {
+            List<List<ItemStack>> input = r.getIngredients();
+            if (input.stream().anyMatch(itemStacks -> itemStacks.stream().anyMatch(inputItem))) {
+                addBackup(r);
+                return true;
+            }
+            return false;
+        });
+    }
+
+    public boolean removeByFluidOutput(FluidStack fluidStack) {
+        return getRecipeList().removeIf(r -> {
+            List<FluidStack> outputFluid = r.getFluidOutputs();
+            if (outputFluid.stream().anyMatch(fluidOut -> fluidOut.isFluidEqual(fluidStack))) {
+                addBackup(r);
+                return true;
+            }
+            return false;
+        });
+    }
+
+    public boolean removeByOutput(IIngredient outputItem) {
+        if (outputItem instanceof FluidStack) {
+            return removeByFluidInput((FluidStack) outputItem);
+        }
+        return getRecipeList().removeIf(r -> {
+            List<ItemStack> output = r.getOutput();
+            if (output.stream().anyMatch(outputItem)) {
+                addBackup(r);
+                return true;
+            }
+            return false;
+        });
+    }
+
+    @MethodDescription(example = @Example(priority = 2000, commented = true))
+    public void removeAll() {
+        List<IRecipe> recipes = getRecipeList();
+        recipes.forEach(this::addBackup);
+        recipes.clear();
+    }
+
+    @MethodDescription(type = MethodDescription.Type.QUERY)
+    public SimpleObjectStream<IRecipe> streamRecipes() {
+        List<IRecipe> recipes = getRecipeList();
+        return new SimpleObjectStream<>(recipes)
+            .setRemover(r -> {
+                if (recipes.remove(r)) {
+                    addBackup(r);
+                    return true;
+                }
+                return false;
+            });
+    }
+
+    @Property(property = "input", valid = {@Comp(type = Comp.Type.GTE, value = "1"), @Comp(type = Comp.Type.LTE, value = "9")})
+    @Property(property = "output", valid = {@Comp(type = Comp.Type.GTE, value = "1"), @Comp(type = Comp.Type.LTE, value = "9")})

```

Comment:
you should be able to set `value = "groovyscript.wiki.advancedrocketry.output.value"` here and have it override this in all the extending classes.